google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplTest.java

```diff
@@ -3419,6 +3419,48 @@ public void testRetryOnResourceExhausted() {
     }
   }
 
+  @Test
+  public void testSampleRetrySettings() {
```

Comment:
Can we carve it into a separate test class similar to what we do for most other samples code? It improves the discoverability of the test.

Reply:
Nit: Consider adding the new test in a separate class. `google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplTest.java` is already overloaded for testing too many things.

Reply:
Moved the sample test to a separate class (and created an abstract base class for creating mock server tests, so it's easier to set up more of those test classes).